File: Data Structures & Algorithms/top-k-elements-in-list/submission-7.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer,Integer> count = new HashMap<>();

        for(int index = 0; index < nums.length; index++){
            count.put(nums[index],count.getOrDefault(nums[index],0) + 1);
        }

        List<Integer>[] bucketArr = new List[nums.length + 1];
        for(int index = 0; index < bucketArr.length; index++){
            bucketArr[index] = new ArrayList();
        }
        for(Map.Entry<Integer,Integer> mapIterate : count.entrySet()){
            bucketArr[mapIterate.getValue()].add(mapIterate.getKey());
        }
        
        int pos = 0;
        int[] answer = new int[k];

        for(int index = bucketArr.length - 1; index > 0; index--){
            for (int num : bucketArr[index]){
                answer[pos++] = num;
                if (pos == k) return answer;
            }
        } 
        return answer;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-7.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # 2 20 4 10 3 4 5
        # 

        checker = set()

        for num in nums:
            checker.add(num)
        
        longestConsecutive = 0
        
        for num in nums:
            consecutive = 1
            if num-1 in checker:
                continue
            else:
                i = num + 1
                while i in checker:
                    i+= 1
                    consecutive += 1
            longestConsecutive = max(longestConsecutive,consecutive)
        return longestConsecutive
            

            